src/libstd/keyword_docs.rs

```diff
@@ -33,9 +33,71 @@ mod as_keyword { }
 //
 /// Exit early from a loop.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
+/// When `break` is encountered, execution of the associated loop body is
+/// immediately terminated.
+///
+/// ```rust
+/// let mut last = 0;
+///
+/// for x in 1..100 {
+///     if x > 12 {
+///         break;
+///     }
+///     last = x;
+/// }
+///
+/// assert_eq!(last, 12);
+/// println!("{}", last);
+/// ```
+///
+/// A break expression is normally associated with the innermost loop enclosing the
+/// `break` but a label can be used to specify which enclosing loop is affected.
+///
+///```rust
+/// 'outer: for i in 1..=5 {
+///     println!("outer iteration (i): {}", i);
+///
+///     'inner: for j in 1..=200 {
+///         println!("    inner iteration (j): {}", j);
+///         if j >= 3 {
+///             // breaks from inner loop, let's outer loop continue.
+///             break;
+///         }
+///         if i >= 2 {
+///             // breaks from outer loop, and directly to "Bye".
+///             break 'outer;
+///         }
+///     }
+/// }
+/// println!("Bye.");
+///```
+///
+/// When associated with `loop`, but not with any other kind of loop expression,
+/// `break` can return a value.  When no value is specified, `break;` returns `()`.
+/// Every `break` within a loop must return the same type.
```

Comment:
I've not done this as a suggestion because that doesn't work with the beta multi-line comment stuff.

I kinda feel like `break` doesn't "return" a value per-se, it's given a value for the `loop` to return.  As such, perhaps the following might be better wording?

```
/// When associated with a `loop` expression, but not with any other iterative expression,
/// `break` can be given a value to return.  When no value is given, `break;` returns `()`.
/// Every `break` within a `loop` must be given values of the same type.
```